pkg/util/pki/csr.go

```diff
@@ -366,18 +369,18 @@ func GenerateTemplateFromCertificateRequest(cr *v1.CertificateRequest) (*x509.Ce
 	if err != nil {
 		return nil, err
 	}
-	return GenerateTemplateFromCSRPEMWithUsages(cr.Spec.Request, certDuration, cr.Spec.IsCA, keyUsage, extKeyUsage)
+	return GenerateTemplateFromCSRPEMWithUsages(cr.Spec.Request, certDuration, cr.Spec.IsCA, cr.Spec.MaxPathLen, keyUsage, extKeyUsage)
 }
 
-func GenerateTemplateFromCSRPEM(csrPEM []byte, duration time.Duration, isCA bool) (*x509.Certificate, error) {
+func GenerateTemplateFromCSRPEM(csrPEM []byte, duration time.Duration, isCA bool, maxPathLen *int) (*x509.Certificate, error) {
```

Comment:
suggestion: As above, this would be a breaking change for anyone using util/pki and it'd be nice to avoid that.

I think these functions are starting to get too complicated with too many arguments and too many variations (and I realise I've added to that). Maybe a `type GenerateTemplateFromCSROptions struct` is the way to go?

Reply:
This has been fixed in a separate PR.